core/admin/mailu/sso/views/base.py

```diff
@@ -57,3 +59,36 @@ def logout():
     flask.session.destroy()
     return flask.redirect(flask.url_for('.login'))
 
+
+@sso.route('/proxy', methods=['GET'])
+@sso.route('/proxy/<target>', methods=['GET'])
+def proxy(target='webmail'):
+    ip = ipaddress.ip_address(flask.request.remote_addr)
+    if not any(ip in cidr for cidr in app.config['PROXY_AUTH_WHITELIST']):
+        return flask.abort(500, '%s is not on PROXY_AUTH_WHITELIST' % flask.request.remote_addr)
+
+    email = flask.request.headers.get(app.config['PROXY_AUTH_HEADER'], None)
+    if not email:
+        return flask.abort(500, 'No %s header' % app.config['PROXY_AUTH_HEADER'])
+
+    user = models.User.get(email)
+    if user:
+        flask.session.regenerate()
+        flask_login.login_user(user)
+        return flask.redirect(app.config['WEB_ADMIN'] if target=='admin' else app.config['WEB_WEBMAIL'])
+    elif app.config['PROXY_AUTH_CREATE']:
+        localpart, desireddomain = email.split('@')
+        domain = models.Domain.query.get(desireddomain) or flask.abort(500, 'You don\'t exist. Go away! (domain=%s)' % desireddomain)
+        if not domain.max_users == -1 and len(domain.users) >= domain.max_users:
```

Comment:
Do we really use `not dmax == -1` in our code.
Shouldn't we use `dmax >= 0` - seems more logical to me.

Reply:
We do; I've copied the pattern we use elsewhere.

```
$rgrep max_users core/admin/ 2> /dev/null
core/admin/migrations/versions/ff0417f4318f_.py:        sa.Column('max_users', sa.Integer(), nullable=False),
core/admin/migrations/versions/fc099bd15cbe_.py:"""change default unlimited value for max_users and max_aliases to -1
core/admin/migrations/versions/fc099bd15cbe_.py:    sa.Column('max_users', sa.Integer(), nullable=False),
core/admin/migrations/versions/fc099bd15cbe_.py:                domain_table.c.max_users == 0
core/admin/migrations/versions/fc099bd15cbe_.py:                    max_users=-1
core/admin/migrations/versions/fc099bd15cbe_.py:        batch.alter_column('max_users', server_default='-1')
core/admin/migrations/versions/fc099bd15cbe_.py:                domain_table.c.max_users == -1
core/admin/migrations/versions/fc099bd15cbe_.py:                    max_users=0
core/admin/migrations/versions/fc099bd15cbe_.py:        batch.alter_column('max_users', server_default='0')
core/admin/mailu/sso/views/base.py:        if not domain.max_users == -1 and len(domain.users) >= domain.max_users:
core/admin/mailu/manage.py:def domain(domain_name, max_users=-1, max_aliases=-1, max_quota_bytes=0):
core/admin/mailu/manage.py:        domain = models.Domain(name=domain_name, max_users=max_users,
core/admin/mailu/manage.py:        max_users = domain_config.get('max_users', -1)
core/admin/mailu/manage.py:                                   max_users=max_users,
core/admin/mailu/manage.py:            domain.max_users = max_users
core/admin/mailu/manage.py:@click.argument('max_users')
core/admin/mailu/manage.py:def setlimits(domain_name, max_users, max_aliases, max_quota_bytes):
core/admin/mailu/manage.py:    domain.max_users = max_users
Binary file core/admin/mailu/__pycache__/manage.cpython-310.pyc matches
core/admin/mailu/ui/forms.py:    max_users = fields_.IntegerField(_('Maximum user count'), [validators.NumberRange(min=-1)], default=10)
core/admin/mailu/ui/templates/user/signup_domain.html:  <td data-sort="{{ -1 if domain.max_users == -1 else domain.max_users - (domain.users | count)}}">{{ '∞' if domain.max_users == -1 else domain.max_users - (domain.users | count)}}</td>
core/admin/mailu/ui/templates/domain/list.html:    <td data-order="{{ domain.users | count }}">{{ domain.users | count }} / {{ '∞' if domain.max_users == -1 else domain.max_users }}</td>
core/admin/mailu/ui/templates/domain/create.html:  {{ macros.form_fields((form.max_users, form.max_aliases)) }}
core/admin/mailu/ui/views/domains.py:            domain.max_users = 10
core/admin/mailu/ui/views/users.py:    if not domain.max_users == -1 and len(domain.users) >= domain.max_users:
core/admin/mailu/ui/views/users.py:        if domain.max_users == -1 or len(domain.users) < domain.max_users
core/admin/mailu/models.py:    max_users = db.Column(db.Integer, nullable=False, default=-1)
```

Reply:
So be it.